aptos-move/block-executor/src/code_cache.rs

```diff
@@ -44,28 +44,22 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> ModuleCodeB
     type Extension = AptosModuleExtension;
     type Key = ModuleId;
     type Verified = Module;
-    type Version = Option<TxnIndex>;
 
     fn build(
         &self,
         key: &Self::Key,
-    ) -> VMResult<
-        Option<ModuleCode<Self::Deserialized, Self::Verified, Self::Extension, Self::Version>>,
-    > {
+    ) -> VMResult<Option<ModuleCode<Self::Deserialized, Self::Verified, Self::Extension>>> {
         let key = T::Key::from_address_and_module_name(key.address(), key.name());
         self.get_raw_base_value(&key)
             .map_err(|err| err.finish(Location::Undefined))?
             .map(|state_value| {
-                let extension = AptosModuleExtension::new(state_value);
+                let extension = Arc::new(AptosModuleExtension::new(state_value));
+
+                // TODO(loader_v2): This recomputes module hash twice, we should avoid it.
```

Comment:
which part computes it twice?

Reply:
Runtime environment when deserialising bytes into compiled module also returns hash and size. We should not do that because now this is stored as "extension".

Reply:
maybe add that context, just in case someone else was fixing it in the future :)

Reply:
I will fix it😄 let me put a PR for that :))

Reply:
Done in #15256